.github/actions/setup-docker/action.yml

```diff
@@ -0,0 +1,35 @@
+---
+name: Setup Docker
+description: Setup Docker Buildx and container registry.
+
+inputs:
+  registry_domain:
+    description: The Docker container registry domain.
+    required: true
+  registry_username:
+    description: The Docker container registry username.
+    required: true
+  registry_password:
+    description: The Docker container registry password.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Setup cache
+      uses: actions/cache@v3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
```

Comment:
I think the build and push action now has native support for this type of caching